data_safe_haven/context_infrastructure/infrastructure.py

```diff
@@ -10,6 +10,12 @@ def __init__(self, context: Context) -> None:
         self.azure_api_: AzureApi | None = None
         self.context = context
         self.tags = {"component": "context"} | context.tags
+        try:
+            if self.azure_api.subscription_id:
+                pass
+        except Exception as exc:
+            msg = "Not logged in or login has expired. \nAuthenticate using az login before creating or tearing down context infrastructure."
+            raise DataSafeHavenAzureError(msg) from exc
```

Comment:
I don't think it is a good idea to do this in the `__init__`, especially as `self.azure_api` is created lazily.

Does this do the same as 

https://github.com/alan-turing-institute/data-safe-haven/blob/5596a995f03be85d30f4c7dd3bdae5b818ddf11a/data_safe_haven/external/api/azure_cli.py#L65-L81

If not, could we add a method there?